### DISCUSSION TITLE
Close notifier window manually via script?

### DISCUSSION THREAD

**Discussion by unsoluble** [IBM/mac-ibm-notifications#190] [Q&A] [answered]
Hello!

What's the best/elegant way to close a `popup` via the triggering script? I've got an indeterminate progress bar that I'd like to make disappear when the underlying action is complete, and not wait for an OK press. Killing the Notifier process doesn't seem to do this.

Thanks!

**Comment by unsoluble** [comment]
Somewhat related followup: When adding a `timer` accessory, with its associated payload, it appears to only ever do the secondary button's action when the time is up. How do I make it fire the `main_button`?

Edit: Ah, okay, found the bit about `timer` returning `4`, that part's all sorted. Would still like to know how to kill the dialog tho. :)

**Comment by SMartorelli** [comment]
Hi @unsoluble, killing the IBM Notifier process should automatically end all its processes and so the pop-up UI itself. You can also consider to use the optional progress bar payload key `/exit_on_completion true` that will make the process to exit when the `end` is sent to the progress bar updates.
E.g.: `"/percent indeterminate /exit_on_completion true"`

**Reply by unsoluble** [reply]
Thanks Simone! I'm still not having any luck with having the window disappear on the PID kill, but I'm more interested now in using this `end` technique — how do I send this to the notifier? Can't find it mentioned in any of the examples or guides.

**Reply by SMartorelli** [reply, marked as answer]
To update the progress bar accessory view, simply continue sending information to the IBM Notifier input pipe. The information must have the same structure as illustrated in the wiki for creating an accessory view (e.g. `/percent 50` - `/percent 90 /top_message Almost done!` - `end` or `/percent 100`). You can refer to [this](https://github.com/IBM/mac-ibm-notifications/wiki/Script-Examples#progress-bar-with-cancel-button) section of the Wiki, which demonstrates the implementation of a standard-defined progress bar. For the indeterminate progress bar, the only distinction is that it recognizes only `/percent 100` and `end` as the "final" updates.

**Reply by unsoluble** [reply]
Ahhh, got it, thanks! I didn't see `end` as recognized syntax anywhere and wasn't clear that an indeterminate bar would accept 100%. Tested and works, all good. :)
